### PR TITLE
VS Code templates for dev & debug on native environments

### PR DESCRIPTION
**Pull request by mocsharp**
This PR adds a tutorial with templates to set up environments for developing and debugging Holoscan applications in native environments (outside of containers).

- A consolidated `tutorials/debugging` directory with all debugging options from PRs #425, #430, #433
- Tutorial on how to build Holoscan SDK in debug mode and how to step into Holoscan SDK source code

**C++**
- A template based on CMake using CMake VS Code extension
- Hello World sample application in C++
- A simple test case in CMake that can be used with CTest and CMake VC Code extension
- Tutorial on how to use the template, configure the project, run and debug the application, and how to run the tests

**Python**
- A template with a `requirements.txt` file to install the Holoscan wheel and pytest
- Hello World sample application in Python
- A simple test case with pytest
- Tutorial on how to use the template, run and debug the application, and how to run the test case

### CODE REVIEW

File: tutorials/debugging/vscode-native/README.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  # clone the Holohub repository
+  git clone https://github.com/nvidia-holoscan/holohub.git
+  ```
+
+### C++ Prerequisites
+
+To build and debug your C++ applications natively, we must first install the redistribution of the Holoscan SDK. To install Holoscan SDK using a Debian package, refer to the Debian package installation section in the [Holoscan SDK documentation](https://docs.nvidia.com/holoscan/sdk-user-guide/sdk_installation.html#instructions).
+
+- [CMake Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools) extension for VS Code
+- [CMake](https://cmake.org/cmake/help/latest/command/install.html) 3.20 or higher
```
Comment (tbirdso) [inline tutorials/debugging/vscode-native/README.md:24]: Is there a reason to use CMake 3.20? We distribute 3.24.0 in the Holoscan SDK dev container

File: tutorials/debugging/vscode-native/README.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+- Ninja 1.10 or higher
+  ```bash
+  sudo apt update && sudo apt install ninja-build
+  ```
+- gcc 11.4 or higher
+  ```bash
+  sudo apt update && sudo apt install build-essential
+  ```
+
+### Python Preriquisites
```
Comment (tbirdso) [inline tutorials/debugging/vscode-native/README.md:34]:
```suggestion
### Python Prerequisites
```

File: tutorials/debugging/vscode-native/README.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+- [Python C++ Debugger](https://marketplace.visualstudio.com/items?itemName=benjamin-simmonds.pythoncpp-debug) extension for VS Code
+- [Python 3.10](https://www.python.org/) or higher
+- [Holoscan SDK PyPI Wheel](https://pypi.org/project/holoscan/)
+  ```bash
+  pip install holoscan
+  ```
+- (Optional) Python virtual environment of your choice
+
+## Environment Set up
```
Comment (tbirdso) [inline tutorials/debugging/vscode-native/README.md:44]:
```suggestion
## Environment Setup
```

File: tutorials/debugging/vscode-native/README.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+```bash
+code tutorials/debugging/vscode-native/python
+
+# Configure a Python virtual environment if desired, e.g.,
+python3 -m venv .venv
+
+# Install dependencies
+pip install -r ./requirements.txt
+```
```
Comment (tbirdso) [inline tutorials/debugging/vscode-native/README.md:164]: Should a user run the latter commands in their standalone shell or in a VSCode terminal? Or does it not matter?

**Review by tbirdso — COMMENTED** [review]
This looks great! Left small comments.

File: tutorials/debugging/vscode-native/README.md (hunk 1 above)
Comment (mocsharp) [reply tutorials/debugging/vscode-native/README.md:24]: Let me change that to `3.24.0`

File: tutorials/debugging/vscode-native/README.md (hunk 4 above)
Comment (mocsharp) [reply tutorials/debugging/vscode-native/README.md:164]: Great point! Let me break it up into two parts.

**Review by tbirdso — APPROVED** [review]
Any concerns about merging now versus with v2.4 release? Will docs links break?

**Comment by mocsharp** [comment]
> Any concerns about merging now versus with v2.4 release? Will docs links break?

Yes. This breaks the link on NGC Holoscan Overview page.

**Comment by tbirdso** [comment]
We've updated the NGC container overview page with the updated link. (Thanks @jjomier )

**Pull request merged**